mlflow/store/artifact/artifact_repo.py

```diff
@@ -147,7 +147,8 @@ def list_artifacts(self, path):
 
     def _is_directory(self, artifact_path):
         listing = self.list_artifacts(artifact_path)
-        return len(listing) > 0
+        is_file = len(listing) == 0 or (len(listing) == 1 and not listing[0].is_dir)
```

Comment:
Should we check if the returned path is same as `artifact_path`? I think the new condition returns `False` for a directory that contains single file.

Reply:
Done!